### PR TITLE
[WEB-8862] Update JS SDK - Cart API endpoints

### PR DESCRIPTION
**Pull request by AnnieCSerato**
Add functions for creating, updating and retrieving a cart on the Ecom Service

I didn't write integration smoke tests for the GET and PUT endpoints, since it seems all the tests check to see if the url **does not** return a 404, whereas these would because it's part of the behaviour of the endpoint. Thus making it redundant to check if the uri works.

### CODE REVIEW

File: src/Ecom.js [hunk 1]
```diff
@@ -197,6 +197,62 @@ import Service from './Service'
...
+ * @property {String} country_code
+ *
+ * @typedef {Object} Promotion
+ * @property {String} description The description for the promotion that is applied to the cart item.
+ * @property {String} [discount_fixed_amount = undefined] The fixed amount discount that is applied to the cart item. It will be presented only if the promotion is `fixed-amount`.
+ * @property {Number} [discount_percentage = undefined] The percentage discount that is applied to the cart item. It will be presented only if the promotion is `percentage`.
+ * @property {String} [end_date = undefined] The date the promotion ends.
+ *
+ * @typedef {Object} ProductType
+ * @property {CatalogProduct[]} items
```
Comment (mshahidi) [inline src/Ecom.js:219]: `ProductType` doesn't have `items` property.

File: src/Ecom.js [hunk 2]
```diff
@@ -197,6 +197,62 @@ import Service from './Service'
...
+ *
+ * @typedef {Object} CartItem
+ * @property {Number} id
+ * @property {ProductType} product_type
+ * @property {Number} quantity
+ * @property {Number} base_amount
+ * @property {Number} total_amount
+ * @property {Number} tax_amount
+ * @property {Number} tax_rate
+ * @property {Number} [total_amount_after_promotion = undefined] The total amount of the cart item after the promotion is applied. It will be presented only if there is a promotion applied to the cart item.
```
Comment (mshahidi) [inline src/Ecom.js:234]: It seems we forgot to remove this from the schema on Ecom service. We may need it in the future but at the moment there is no implementation for it. Please remove this and I will make a PR on Ecom service to remove it from schema too.

File: src/Ecom.js [hunk 3]
```diff
@@ -197,6 +197,62 @@ import Service from './Service'
  *
  * @typedef {Object} RecommendationsList
  * @property {CatalogProduct[]} items
+ *
+ * @typedef {Object} BillingAddress
```
Comment (mshahidi) [inline src/Ecom.js:201]: I suggest we call this `CartBillingAddress`. We will have `CustomerBillingAddress` which is slightly different with the cart one.

**Review by mshahidi — COMMENTED** [review]
Good job @AnnieCSerato 
I added 3 comments. 
Also FYI, I think we will have some minor changes in these endpoints in the future that should be reflected in the SDK.

File: src/Ecom.js [hunk 4]
```diff
@@ -556,4 +610,75 @@ export default class EcomService extends Service {
...
+   *
+   * @param {Object} param Options
+   * @param {String} param.cartId
+   * @param {Omit<CartBillingAddress, 'country_code'>} param.billingAddress
+   * @return {Promise<Cart>}
+   */
+  updateCartBillingAddress ({ cartId, billingAddress }) {
+    return this.fetch(
+      this._sws.accessToken ? this.bearerTokenAuthHeader() : null,
+      '/api/v1/carts/' + cartId,
```
Comment (sameeraroshan-serato) [inline src/Ecom.js:658]: shall we use template literals here /api/v1/carts/${cartId}

**Review by sameeraroshan-serato — APPROVED** [review]
Nice Work @AnnieCSerato . Small suggestion added.

**Review by mshahidi — APPROVED** [review]

**Pull request merged**